packages/automerge-repo/src/DocHandle.ts

```diff
@@ -78,156 +63,93 @@ export class DocHandle<T> //
-          states: {
-            idle: {
-              on: {
-                // If we're creating a new document, we don't need to load anything
-                CREATE: { target: READY },
-                // If we're accessing an existing document, we need to request it from storage
-                // and/or the network
-                FIND: { target: LOADING },
-                DELETE: { actions: "onDelete", target: DELETED },
-              },
-            },
-            loading: {
-              on: {
-                // UPDATE is called by the Repo if the document is found in storage
-                UPDATE: { actions: "onUpdate", target: READY },
-                // REQUEST is called by the Repo if the document is not found in storage
-                REQUEST: { target: REQUESTING },
-                // AWAIT_NETWORK is called by the repo if the document is not found in storage but the network is not yet ready
-                AWAIT_NETWORK: { target: AWAITING_NETWORK },
-                DELETE: { actions: "onDelete", target: DELETED },
-              },
-              after: [
-                {
-                  delay: this.#timeoutDelay,
-                  target: UNAVAILABLE,
-                },
-              ],
-            },
-            awaitingNetwork: {
-              on: {
-                NETWORK_READY: { target: REQUESTING },
-              },
-            },
-            requesting: {
-              on: {
-                MARK_UNAVAILABLE: {
-                  target: UNAVAILABLE,
-                  actions: "onUnavailable",
-                },
-                // UPDATE is called by the Repo when we receive changes from the network
-                UPDATE: { actions: "onUpdate" },
-                // REQUEST_COMPLETE is called from `onUpdate` when the doc has been fully loaded from the network
-                REQUEST_COMPLETE: { target: READY },
-                DELETE: { actions: "onDelete", target: DELETED },
-              },
-              after: [
-                {
-                  delay: this.#timeoutDelay,
-                  target: UNAVAILABLE,
-                },
-              ],
-            },
-            ready: {
-              on: {
-                // UPDATE is called by the Repo when we receive changes from the network
-                UPDATE: { actions: "onUpdate", target: READY },
-                DELETE: { actions: "onDelete", target: DELETED },
-              },
-            },
-            deleted: {
-              type: "final",
-            },
-            unavailable: {
-              on: {
-                UPDATE: { actions: "onUpdate" },
-                // REQUEST_COMPLETE is called from `onUpdate` when the doc has been fully loaded from the network
-                REQUEST_COMPLETE: { target: READY },
-                DELETE: { actions: "onDelete", target: DELETED },
-              },
-            },
+    const delay = this.#timeoutDelay
+    const machine = setup({
```

Comment:
This `setup(...).createMachine(...)` is XState's new recommended API for associating types and actions with a machine config.